lib/transports/rest2.js

```diff
@@ -250,6 +254,36 @@ class RESTv2 {
     return new ModelClass(data)
   }
 
+  /**
+   * @param {Arr} data order matters
+   * @return {Arr} merged arr of coins and names with no pairs rpeated
+   * @private
+   */
+  _mergeAllArrOfCoins (data) {
+    const addKeyToObj = (obj, curr) => {
+      if (!Array.isArray(curr)) obj[curr] = curr
+      else if (curr.length === 1) obj[curr[0]] = curr[0]
+      else obj[curr[0]] = curr[1]
+
+      return obj
+    }
+
+    const arrOfKeyObjects = _transform(data, (allObjects, arr, ii) => {
+      const keyObject = _transform(arr, (tempObj, curr, i) => {
+        return addKeyToObj(tempObj, curr)
+      }, {})
+
+      return allObjects.push(keyObject)
+    }, [])
+
+    const merge = _assign({}, ...arrOfKeyObjects)
+    const res = _transform(merge, (result, value, index) => {
+      result.push([index, value])
```

Comment:
`result` is an object here, isn't it? (since `merge` is an object.) How can this work?

Reply:
No, result is and accumulator that is an array, had initialize it on last part of the function as an array 
_transform(merge, (result, value, index) => {
}, [] ) // this [] is result